### PR TITLE
Features - Load Virtual Runset

### PR DESCRIPTION
**Pull request by IamRanjeetSingh**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [x] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [x] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [x] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
	- Introduced a new color conversion logic for RunSet labels based on their status.
	- Enhanced the RunSet configuration loading process to be asynchronous, improving the application's responsiveness.
	- Added new user messages to better guide users through RunSet loading errors and confirmations.

- **Enhancements**
	- Improved RunSet information and execution history sections for better clarity and user experience.
	- Implemented a dynamic post-save handler for Business Flows to extend functionality after saving.

- **Bug Fixes**
	- Fixed an issue with RunSet label color not updating correctly by removing hardcoded color values.

### CODE REVIEW

File: Ginger/Ginger/RunSetPageLib/NewRunSetPage.xaml.cs [hunk 1]
```diff
@@ -51,6 +51,7 @@
 using System.Collections.Generic;
 using System.Collections.Specialized;
 using System.ComponentModel;
+using System.Globalization;
```
[the remark this answers is not in the document]
Comment (IamRanjeetSingh) [reply Ginger/Ginger/RunSetPageLib/NewRunSetPage.xaml.cs:54]: yes, it is needed.

File: Ginger/GingerCoreNET/RunLib/DynamicExecutionLib/DynamicExecutionManager.cs [hunk 2]
```diff
@@ -1638,8 +1637,7 @@ public static void CreateUpdateRunSetFromJSON(RunsetExecutor runsetExecutor, Gin
                 runSetConfig.AllowInterActivityFlowControls = (bool)dynamicRunsetConfigs.AllowInterActivityFlowControls;
             }
 
-            // Set config
-            runsetExecutor.RunSetConfig = runSetConfig;
+            return runSetConfig;
```
[the remark this answers is not in the document]
Comment (IamRanjeetSingh) [reply Ginger/GingerCoreNET/RunLib/DynamicExecutionLib/DynamicExecutionManager.cs:1640]: logging is handled in other parts of the code

File: Ginger/GingerCoreNET/WorkSpaceLib/SolutionOperations.cs [hunk 3]
```diff
@@ -366,5 +367,19 @@ public static ObservableList<SolutionCategoryValue> GetSolutionReleaseValues()
 
         }
 
+        public void CleanUpCacheDirectory()
+        {
+            Type[] repositoryItemTypes = [typeof(RunSetConfig), typeof(BusinessFlow)];
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/WorkSpaceLib/SolutionOperations.cs:372]: Add try catch

File: Ginger/GingerCoreNET/Run/RemoteExecution/IExecutionHandlerAPIClient.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using static Amdocs.Ginger.CoreNET.Run.RemoteExecution.ExecutionHandlerAPIClient;
+
+#nullable enable
+namespace Amdocs.Ginger.CoreNET.Run.RemoteExecution
+{
+    public interface IExecutionHandlerAPIClient : IDisposable
+    {
+        public Task<ExecutionDetailsResponse?> GetExecutionDetailsAsync(string executionId, ExecutionDetailsOptions options);
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Run/RemoteExecution/IExecutionHandlerAPIClient.cs:14]: Add Interface for Execute API

File: Ginger/GingerCoreNET/Reports/RunsetFromReportLoader.cs [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+using System.Threading.Tasks;
+
+#nullable enable
+namespace Amdocs.Ginger.CoreNET.Reports
+{
+    public sealed class RunsetFromReportLoader
+    {
+        private readonly JsonSerializerOptions GingerExecConfigSerializationOptions;
+
+        public sealed class RunsetLoadResult
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Reports/RunsetFromReportLoader.cs:25]: remove if not used

File: Ginger/GingerCoreNET/Reports/RunsetFromReportLoader.cs [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+        {
+            GingerExecConfigSerializationOptions = new();
+            GingerExecConfigSerializationOptions.Converters.Add(new JsonStringEnumConverter());
+        }
+
+        public async Task<RunsetLoadResult> LoadAsync(RunSetReport runsetReport)
+        {
+            string runsetName = runsetReport.Name;
+            RunSetConfig? runset = GetRunsetFromSolutionRepository(runsetName);
+            bool isVirtual = false;
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Reports/RunsetFromReportLoader.cs:42]: remove as it is not needed

File: Ginger/GingerCoreCommon/Repository/SolutionRepository.cs [hunk 7]
```diff
@@ -564,7 +567,7 @@ private void VerifyOrCreateSolutionFolders(string folder)
             }
         }
 
-        internal void SaveNewRepositoryItem(RepositoryItemBase repositoryItem)
+        public void SaveNewRepositoryItem(RepositoryItemBase repositoryItem)
```
Comment (Maheshkale447) [inline Ginger/GingerCoreCommon/Repository/SolutionRepository.cs:570]: Keep it as internal

File: Ginger/GingerCoreCommon/Repository/SolutionRepository.cs [hunk 8]
```diff
@@ -53,6 +53,9 @@ public class SolutionRepository
              @"ExecutionResults\",
              @"HTMLReports\",
 
+             $@"BusinessFlows\{ISolution.CacheDirectoryName}",
```
Comment (Maheshkale447) [inline Ginger/GingerCoreCommon/Repository/SolutionRepository.cs:56]: Enhance it to update existing git ignore files

File: Ginger/GingerCoreCommon/ReporterLib/UserMsgsPool.cs [hunk 9]
```diff
@@ -468,7 +473,7 @@ public static void LoadUserMsgsPool()
...
             Reporter.UserMsgsPool.Add(eUserMsgKey.UpdateApplicationNameChangeInSolution, new UserMsg(eUserMsgType.WARN, $"{GingerDicser.GetTermResValue(eTermResKey.TargetApplication)} Name Change", $"Do you want to automatically update the {GingerDicser.GetTermResValue(eTermResKey.TargetApplication)} name in all Solution items?" + Environment.NewLine + Environment.NewLine + "Note: If you choose 'Yes', changes won't be saved, for saving them please click 'SaveAll'", eUserMsgOption.YesNo, eUserMsgSelection.Yes));
-            Reporter.UserMsgsPool.Add(eUserMsgKey.SaveRunsetChanges, new UserMsg(eUserMsgType.QUESTION, "Save Changes", "There are unsaved changes in runset, Do you want to save it?", eUserMsgOption.YesNo, eUserMsgSelection.No));
+            Reporter.UserMsgsPool.Add(eUserMsgKey.SaveRunsetChanges, new UserMsg(eUserMsgType.QUESTION, "Save Changes", $"There are unsaved changes in runset, Do you want to save it?{Environment.NewLine}{Environment.NewLine}Notice: {ISolution.CacheDirectoryName} folder files will be lost after solution reload.", eUserMsgOption.YesNo, eUserMsgSelection.No));
```
Comment (Maheshkale447) [inline Ginger/GingerCoreCommon/ReporterLib/UserMsgsPool.cs:476]: change Notice to Note

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**